### PR TITLE
Standard tests documentation & docs/template improvements

### PR DESCRIPTION
**Pull request by sherifnada**
## What
* Rewrite parts of the "Building New Connectors" section to make it hopefully better at conveying the higher level picture of what it required to add a new connector
* Adds basic documentation for what standard tests are, how they work, and how to configure the simplest one in a new connector
* Adds a template generator for a "generic" connector i.e: one which doesn't have any other template

### CODE REVIEW

File: airbyte-integrations/connector-templates/source-generic/NEW_SOURCE_CHECKLIST.md.hbs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# New generic source checklist
+
+This is an autogenerated file describing the steps needed to implement a new Airbyte source.
```
Comment (sherifnada) [inline airbyte-integrations/connector-templates/source-generic/NEW_SOURCE_CHECKLIST.md.hbs:3]: made this one much shorter than the average checklist. I think from now on we should put code-specific action items as TODOs in the code, and cover more conceptual topics in the docs. That's the idea anyways.

File: docs/contributing-to-airbyte/building-new-connector/testing-connectors.md [hunk 2]
```diff
@@ -1,16 +1,80 @@
...
+  configPath = "/path/to/config"
+  specPath = "/path/to/spec"
+  configuredCatalogPath = "/path/to/catalog"
+}
+```
+
+These inputs are all described in the [Airbyte Specification](../../architecture/airbyte-specification.md) and will be used as follows: 
+* **Spec file** will be compared to the spec file output by the connector when the `spec` command is called. 
+* **Config file** is expected to be a valid config file. It's expected that calling `check` with this config will succeed. 
+* **Configured Catalog** read operations will be performed on all the streams found in this catalog. All sync modes supported for each stream will be tested. If any stream requires a user-defined cursor, this should be defined in this catalog file. (If this sounds like gibberish, make sure to read about incremental sync).
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/building-new-connector/testing-connectors.md:52]: TODO once I add docs for incremental sync i'll link to them in the part that says "make sure to read about incremental sync"

File: docs/contributing-to-airbyte/building-new-connector/testing-connectors.md [hunk 3]
```diff
@@ -1,16 +1,80 @@
...
 
+
+## Airbyte's Standard Tests
+To ensure a minimum quality bar, Airbyte runs all connectors against the same set of integration tests (sources & destinations 
+have two different test suites). Those tests ensure that each connector adheres to the 
+[Airbyte Specification](../../architecture/airbyte-specification.md) and responds correctly to Airbyte commands when provided valid 
+(or invalid) inputs. 
+
+### Architecture of standard tests
+![Standard test sequence diagram](../../.gitbook/assets/standard_tests_sequence1.png)
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/building-new-connector/testing-connectors.md:11]: make sure to view in pretty mode to see the diagram

File: docs/contributing-to-airbyte/building-new-connector/README.md [hunk 4]
```diff
@@ -1,47 +1,54 @@
 # Building New Connectors
 
-A connector takes the form of a Docker image which follows the [Airbyte specification](../../architecture/airbyte-specification.md).
+Airbyte supports 2 types of connectors: Sources and Destinations. A connector takes the form of a Docker image which follows the [Airbyte specification](../../architecture/airbyte-specification.md).
```
Comment (jrhizor) [inline docs/contributing-to-airbyte/building-new-connector/README.md:3]:
```suggestion
Airbyte supports two types of connectors: Sources and Destinations. A connector takes the form of a Docker image which follows the [Airbyte specification](../../architecture/airbyte-specification.md).
```

File: docs/contributing-to-airbyte/building-new-connector/README.md [hunk 5]
```diff
@@ -1,47 +1,54 @@
...
-**Please note that, at no point in time, we will ask you to maintain your connector.** The goal is that the Airbyte team and the community helps maintain the connector.
+**Note: you are not required to maintain the connectors you create.** The goal is that the Airbyte core team and the community help maintain the connector.
 
 ## The Airbyte specification
 
-Before you can start building your own connector, you need to understand [Airbyte's data protocol specification](../../architecture/airbyte-specification.md).
+Before building a new connector, review [Airbyte's data protocol specification](../../architecture/airbyte-specification.md).
 
-## Creating a new connector
+## Adding a new connector
```
Comment (jrhizor) [inline docs/contributing-to-airbyte/building-new-connector/README.md:13]:
```suggestion
## Adding a new connector
```

File: docs/contributing-to-airbyte/building-new-connector/README.md [hunk 6]
```diff
@@ -1,47 +1,54 @@
...
+Before building a new connector, review [Airbyte's data protocol specification](../../architecture/airbyte-specification.md).
 
-## Creating a new connector
+## Adding a new connector 
 
-First, make sure you built the project by running
+### Requirements
+To add a new connector you need to: 
+1. Package your connector in an Airbyte Protocol compliant Docker image
+2. Add integration tests for your connector. At a minimum, all connectors must pass Airbyte's standard test suite, but you can also add your own tests.
```
Comment (jrhizor) [inline docs/contributing-to-airbyte/building-new-connector/README.md:18]: `standard test suite` should link to those docs.

File: docs/contributing-to-airbyte/building-new-connector/README.md [hunk 7]
```diff
@@ -1,47 +1,54 @@
...
+3. Add the appropriate Gradle tasks to build the image within the Airbyte monorepo and CI
+
+Each requirement has a subsection below.
```
Comment (jrhizor) [inline docs/contributing-to-airbyte/building-new-connector/README.md:21]: The headers below should be numbered to match the summaries above.

File: docs/contributing-to-airbyte/building-new-connector/README.md [hunk 8]
```diff
@@ -1,47 +1,54 @@
...
-
-The generated `README.md` will also contain instructions on how to iterate.
-
-### Creating a Connector from a Singer Tap
-
-[Singer.io](http://singer.io/) is an established open source framework with a vibrant community and many available connectors \(known as taps & targets\). Airbyte makes it simple & easy to build an Airbyte connector from a Singer tap. The basic idea is to wrap the Singer tap in a shim Python package which makes the Singer tap compatible with the Airbyte protocols, then publishing that in a Docker image consumable by Airbyte.
+and choose the relevant template. This will generate a new connector in the `airbyte-integrations/connectors/<your-connector>` directory.
 
-To create an Airbyte connector from a Singer tap, run the connector generator as described above and choose the Singer generator. The generated package will contain the details about exactly what needs to be done.
+Follow the instructions generated in the `NEW_SOURCE_CHECKLIST.md` file to bootstrap the connector.
```
Comment (jrhizor) [inline docs/contributing-to-airbyte/building-new-connector/README.md:46]:
```suggestion
Follow the instructions generated in the `NEW_SOURCE_CHECKLIST.md` file to complete the connector.
```

The templater is bootstrapping, nothing after that point is bootstrapping.

File: docs/contributing-to-airbyte/building-new-connector/README.md [hunk 9]
```diff
@@ -1,47 +1,54 @@
...
 
-See the [Github Connector](https://github.com/airbytehq/airbyte/tree/master/airbyte-integrations/connectors/source-github-singer) for an example of an Airbyte Connector implemented on top of a Singer tap.
+### Integrating with Gradle
+All generated templates should provide this automatically, but generally speaking to integrate with Gradle your connector needs the following:
```
Comment (jrhizor) [inline docs/contributing-to-airbyte/building-new-connector/README.md:49]:
```suggestion
Generated templates provide the following Gradle tasks: 
```

**Review by jrhizor — APPROVED** [review]

File: docs/contributing-to-airbyte/building-new-connector/README.md (hunk 7 above)
Comment (sherifnada) [reply docs/contributing-to-airbyte/building-new-connector/README.md:21]: done

**Pull request merged**